docker-compose.override.yml

```diff
@@ -41,24 +38,39 @@ services:
       done &&
       exec $$0 \"$$@\""
     command: minio server /data --address :5000 --console-address :5001
-    volumes:
-      # Any buckets used for testing provider data imported from s3 should be subdirectories under 
-      # /tests/s3-data/
-      # The trick to the issues here (https://wordpress.slack.com/archives/C02012JB00N/p1655118749223179)
-      # was to have a single volume, and to represent each bucket as a subdirectory under that.
-      # BUT, the test runs still can't see the inaturalist-open-data bucket, even though I have 
-      # verified through docker that the files exist on the container.
-      # TO DO: The openverse user has root access to all buckets on this minio instance, which
-      # makes sense for openverse-owned buckets, but not for read-only buckets owned by providers.
-      # Maybe we add a minio/mc + minio/minio pair with more realistic permissions for buckets
-      # containing sample provider data?
-      - ./tests/s3-data:/data:rw
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:5010/minio/health/live"]
       interval: 30s
       timeout: 20s
       retries: 3
 
+  load_to_s3:
+    image: minio/mc:latest
+    env_file:
+      - .env
+    depends_on:
+      - s3
+    volumes:
+      # Buckets for testing provider data imported from s3 are subdirectories under 
+      # /tests/s3-data/
+      - ./tests/s3-data:/data:rw
+    # Loop through subdirectories mounted to the volume and load them to s3/minio.
+    # This takes care of filesystem delays on some local dev environments that may make
+    # minio miss files included directly in the minio volume.
+    # More info here: https://stackoverflow.com/questions/72867045
+    # This does *not* allow for testing permissions issues that may come up in real AWS.
+    # And, if you remove files from /tests/s3-data, you will need to use `just down -v`
+    # and `just up` or `just recreate` to see the minio bucket without those files.
+    entrypoint: >
+      /bin/sh -c "
+      /usr/bin/mc config host add s3 http://s3:5000 ${AWS_ACCESS_KEY} ${AWS_SECRET_KEY};
+      cd /data;
+      for b in */ ; do
+        echo \"Loading bucket $$b\" && /usr/bin/mc mb --ignore-existing s3/$$b && /usr/bin/mc cp --r $$b s3/$$b && /usr/bin/mc ls s3/$$b;
```

Comment:
Brilliant! I'm so glad this worked! I think it's also overall going to be a more consistent way to accomplish what we were accomplishing before.

Is it possible to split this into multiple lines instead of using `&&`?

Reply:
Ha! Yes, and that's so much better.